python/requirements_compiled.txt

```diff
@@ -220,8 +230,8 @@ jsonschema==4.17.3
 jsonschema-spec==0.1.6
 junit-xml==1.9
 jupyter-cache==0.4.3
-jupyter-client==7.3.4
-jupyter-core==5.3.1
+jupyter-client==7.3.4 ; python_version >= "3.8"
+jupyter-core==5.3.1 ; python_version >= "3.8"
```

Comment:
why are these `python_version >= "3.8"` required? what will happen for python 3.7 here?

Reply:
in particular, is it that for python 3.7 these libraries are just not required/installed?

Reply:
Some of these dependencies have sub-dependencies that are unavailable for Python 3.7. The `python_version >= "3.8"` effectively removes the constraint for python 3.7.

In this PR, I compiled requirements for python 3.7 separately and added the selector for any constraints that were resolved to a lower version than in 3.8. This seems to resolve fine.

This is not a sustainable practice. While we can try to update single dependencies manually in the short run, we do need a comprehensive compilation mechanism.

Reply:
hmm... maybe should just have a separate file for 3.7 given there are so many differences.